### ISSUE TITLE
How to search in all fields?

### ISSUE THREAD

**Issue by ngrilly** [Kronuz/Xapiand#19]
Is it possible to search in all fields when using the SEARCH HTTP method?

To search in a given field, I send a query like this one: `{"_query": {"fieldname": "search terms"}}`.

To search in all fields, I tried this: `{"_query": "search terms"}`. But the query returned no documents.

**Comment by YosefMac** [comment]
Yes!! You can search in all fields, but you have to add `"_index": "terms"` which means you can search adding the field or without the field.

The document must be something like this:
```
{
      "fieldname1": {
            "_type": "keyword",
            "_value": "value1",
            "_index": "terms"
      }
}
```
And the search should work like you tried!

By default the value of `_index` is `field_terms` which force use field in the search

**Comment by ngrilly** [comment]
Hi,

I try added documents specifying the indexing mode like above with `"_index": "terms"` but it doesn't work. I can still find the document using `?q=fieldname1:hello` but not using `?q=hello`.

I don't understand why. 

I've checked the index and `"_index": "terms"` is there:

```
curl http://xapiand:8880/document/?pretty
{
  "_settings": {
    "number_of_shards": 5,
    "number_of_replicas": 1
  },
  "_schema": {
    "_id": {
      "_type": "integer",
      "_slot": 0,
      "_prefix": "Q"
    },
    "text": {
      "_index": "all",
      "_prefix": "text.",
      "_ngram": false,
      "_cjk_ngram": false,
      "_cjk_words": false,
      "_slot": 989045663,
      "_type": "text"
    },
    "_version": {
      "_prefix": "V",
      "_slot": 1,
      "_type": "positive"
    },
    "_type": "foreign/object",
    "_foreign": ".xapiand/indices/document"
  }
}
```

**Comment by YosefMac** [comment]
Hi!! Indeed! this is a problem! without the field we try to guess the type but we have no way of differentiating between text type and keyword type.  So it's being guessed as a keyword. But I think is fixed by 4b44a70. Thanks for report the issue!!

**Comment by ngrilly** [comment]
It looks it works. Thanks!
